.github/workflows/ci.yml

```diff
@@ -115,9 +224,152 @@ jobs:
             echo "One or more lint jobs have failed."
             exit 1
           fi
-          result="${{ needs.project-test-jobs.result }}"
+          result="${{ needs.project-default-test-jobs.result }}"
           if [[ $result != "success" && $result != "skipped" ]]; then
             echo "One or more test jobs have failed."
             exit 1
           fi
-          echo "All jobs have completed successfully."
\ No newline at end of file
+          result="${{ needs.project-e2e-test-jobs.result }}"
+          if [[ $result != "success" && $result != "skipped" ]]; then
+            echo "One or more e2e test jobs have failed."
+            exit 1
+          fi
+          result="${{ needs.project-api-test-jobs.result }}"
+          if [[ $result != "success" && $result != "skipped" ]]; then
+            echo "One or more api test jobs have failed."
+            exit 1
+          fi
+          echo "All jobs have completed successfully."
+
+  publish-e2e-test-reports:
+    name: 'Publish e2e test reports'
+    needs: [ project-e2e-test-jobs ]
+    if: ${{ contains( needs.*.result, 'success' ) || contains( needs.*.result, 'failure' ) }}
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: 'Install Allure CLI'
+        env:
+          DESTINATION_PATH: ../
+        run: ./.github/workflows/scripts/install-allure.sh
+
+      - name: 'Download blob reports from artifacts'
+        uses: actions/download-artifact@v4
+        with:
+          path: ./out/allure-results
+          pattern: all-blob-reports-*
+          run-id: project-e2e-test-jobs
+          merge-multiple: true
+
+      - name: 'Generate Allure report'
+        id: generate_allure_report
+        run: allure generate --clean ./out/allure-results --output ./out/allure-report
+
+      - name: 'Archive reports'
+        uses: actions/upload-artifact@v4
+        with:
+          name: e2e-test-report
+          path: ./out
+          if-no-files-found: ignore
+          retention-days: 5
+          
+      - name: 'Publish reports'
+        env:
+          GITHUB_TOKEN: ${{ secrets.REPORTS_TOKEN }}
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+          RUN_ID: ${{ github.run_id }}
+          COMMIT_SHA: ${{ github.event.pull_request.head.sha }}
+        run: |
+          if [ "$GITHUB_EVENT_NAME" == pull_request ]; then
+            WORKFLOW_NAME=publish-test-reports-pr.yml
```

Comment:
Is `WORKFLOW_NAME` used anywhere?

Reply:
Removed with [`64e4a04` (#45190)](https://github.com/woocommerce/woocommerce/pull/45190/commits/64e4a041d06d99a05526445a6e656808d8b236da)